src/ssl/support.cc

```diff
@@ -423,6 +438,78 @@ Ssl::DisablePeerVerification(Security::ContextPointer &ctx)
     SSL_CTX_set_verify(ctx.get(),SSL_VERIFY_NONE,nullptr);
 }
 
+static int squidX509VerifyCert(X509_STORE_CTX *ctx, STACK_OF(X509) *extraCerts);
+
+bool
+Ssl::PeerCertificatesVerify(Security::SessionPointer &s, const Ssl::X509_STACK_Pointer &extraCerts)
+{
+    STACK_OF(X509) *peerCertificatesChain = SSL_get_peer_cert_chain(s.get());
+
+    if (!peerCertificatesChain || sk_X509_num(peerCertificatesChain) == 0) {
+        debugs(83, 2, "no server certificates?");
+        return false;
+    }
+
+    X509_STORE *verificationStore = SSL_CTX_get_cert_store(SSL_get_SSL_CTX(s.get()));
+    X509_STORE_CTX_Pointer storeCtx(X509_STORE_CTX_new());
+    X509 *peerCert = sk_X509_value(peerCertificatesChain, 0);
+    if (!X509_STORE_CTX_init(storeCtx.get(), verificationStore, peerCert, peerCertificatesChain)) {
+        debugs(83, DBG_IMPORTANT, "error initializing X509_STORE");
+        return false;
+    }
+
+    // Retrieve certificate flags:
+    long certFlags = SSL_set_cert_flags(s.get(), 0);
+    if (certFlags & SSL_CERT_FLAG_SUITEB_128_LOS) {
+        auto SSL_CERT_FLAG_to_X509_V_FLAG = [](long flag) {return flag;};
+        long x509Flags = SSL_CERT_FLAG_to_X509_V_FLAG(certFlags);
+        X509_STORE_CTX_set_flags(storeCtx.get(), x509Flags & X509_V_FLAG_SUITEB_128_LOS);
+    }
+
+    if (!X509_STORE_CTX_set_ex_data(storeCtx.get(), SSL_get_ex_data_X509_STORE_CTX_idx(), s.get())) {
+        debugs(83, DBG_IMPORTANT, "error attaching session object to X509_STORE");
+        return false;
+    }
+
+    // The X509_verify_cert will only use dane if it is enabled.
```

Comment:
This was missed in https://github.com/measurement-factory/squid/pull/104#discussion_r539478830, commit f5fd29a4:

```suggestion
    // If we ever add DANE support to Squid, we will supply DANE details here:
```

Feel free to ignore -- I can fix this during polishing.